### PR TITLE
Moving Keys.hs to Cardano.Ledger from Shelley.Spec.Ledger

### PR DESCRIPTION
**Pull request by TimSheard**
Moved Keys and Serialization into Cardano.Ledger. 
Add new files with the old names, which import the moved old ones.
These modules are Deprecated and they exhort you to import the moved ones.

Next up
Move Basetypes and Address

### CODE REVIEW

**Review by JaredCorduan — APPROVED** [review]
I like the idea of these living in core, thanks!

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs [hunk 1]
```diff
@@ -41,6 +41,7 @@ import Cardano.Ledger.Alonzo.Scripts
     isPlutusScript,
   )
 import Cardano.Ledger.AuxiliaryData (AuxiliaryDataHash (..))
+import Cardano.Ledger.BaseTypes (StrictMaybe (..))
```
Comment (nc6) [inline alonzo/impl/src/Cardano/Ledger/Alonzo/Data.hs:44]: Should import this from `Data.Maybe.Strict` instead

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs [hunk 2]
```diff
@@ -1,390 +1,6 @@
...
-    activeSlotVal,
-    activeSlotLog,
-
-    -- * STS Base
-    Globals (..),
-    epochInfo,
-    ShelleyBase,
-  )
+  {-# DEPRECATED "Use 'import Cardano.Ledger.BaseTypes' instead." #-}
+  (module X)
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs:3]: Does this work?

ETA: Interesting - it seems we can deprecate the module, albeit not the re-export.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Serialization.hs [hunk 3]
```diff
@@ -1,293 +1,6 @@
-{-# LANGUAGE BangPatterns #-}
-{-# LANGUAGE FlexibleContexts #-}
-{-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE GeneralizedNewtypeDeriving #-}
-{-# LANGUAGE LambdaCase #-}
-{-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE Rank2Types #-}
-{-# LANGUAGE ScopedTypeVariables #-}
-
 module Shelley.Spec.Ledger.Serialization
```
Comment (nc6) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Serialization.hs:1]: If we're moving this module, we shouldn't move it to `cardano-ledger-core`, but to https://github.com/input-output-hk/cardano-base/blob/master/binary/src/Cardano/Binary/ instead. Pretty much none of this is ledger specific. We should also review to see how much of it is actually used.

**Review by nc6 — APPROVED** [review]
Keys and BaseTypes seem reasonable. Serialisation should really not live here at all. And it would be nice to turn `BaseTypes` into less of a grab-bag, and instead split it out into sensible modules.

**Pull request merged**